packages/server/lambdas/alfresco/upload-file.ts

```diff
@@ -0,0 +1,76 @@
+import { CategoryDataBase } from '@prisma/client';
+import { ALBEvent, ALBResult } from 'aws-lambda';
+import { isEmpty } from 'lodash';
+import { findEndpoint, getAlfrescoOptions } from '../../utils/alfresco';
+import { getRataExtraLambdaError, RataExtraLambdaError } from '../../utils/errors';
+import { log } from '../../utils/logger';
+import { getUser, validateReadUser, validateWriteUser } from '../../utils/userService';
+import { DatabaseClient } from '../database/client';
+import { fileRequestBuilder } from './fileRequestBuilder';
+import fetch from 'node-fetch';
+import { RequestInit } from 'node-fetch';
+
+const database = await DatabaseClient.build();
+
+const fileEndpointsCache: Array<CategoryDataBase> = [];
+
+const postFile = async (options: RequestInit, nodeId: string) => {
+  const url = `https://api.testivaylapilvi.fi/alfresco/api/-default-/public/alfresco/versions/1/nodes/${nodeId}/children`;
+  try {
+    const res = await fetch(url, options);
+    const result = await res.text();
+    return result;
+  } catch (err) {
+    console.error('error:' + err);
+  }
+};
+
+/**
+ * Update custom content for page. Example request: /api/alfresco/file/linjakaaviot
+ * @param {ALBEvent} event
+ * @param {{string}} event.path Path should end with the page to upload the file to
+ * @param {{string}} event.body File contents and metadata to upload
+ * @returns  {Promise<ALBResult>} JSON stringified object of uploaded file metadata
+ */
+export async function handleRequest(event: ALBEvent): Promise<ALBResult | undefined> {
+  try {
+    const paths = event.path.split('/');
+    const category = paths.pop();
+
+    const user = await getUser(event);
+    validateReadUser(user);
+
+    if (!category || paths.pop() !== 'file') {
+      throw new RataExtraLambdaError('Category missing from path', 400);
+    }
+    if (isEmpty(event.body)) {
+      throw new RataExtraLambdaError('Request body missing', 400);
+    }
+    if (!fileEndpointsCache.length) {
+      const categories = await database.categoryDataBase.findMany();
+      categories.forEach((category) => {
```

Comment:
If this is how it should be, then this should be added to other lambdas as well. Not in this ticket, but in it's own ticket. I originally tested this with only one value available in the DB, so I definitely could have missed this.

Out of interest: Why do we have to forEach this instead of setting the cache as the return value?

Reply:
Good point. We don't need to use loop here 👍🏻